python/robot_properties_solo/config.py

```diff
@@ -55,25 +91,70 @@ class SoloConfig:
     # joints
     nb_joints = robot_model.nv - 6
 
-    # control time period
-    control_period = 0.001
-    dt = control_period
+    # mapping between the ctrl vector in the device and the urdf indexes
+    urdf_to_dgm = (0,1,2,3,4,5,6,7)
 
+    map_joint_name_to_id = {}
+    map_joint_limits = {}
+    for i, (name, lb, ub) in enumerate(zip(robot_model.names[1:],
+                                       robot_model.lowerPositionLimit,
+                                       robot_model.upperPositionLimit)):
+        map_joint_name_to_id[name] = i
+        map_joint_limits[i] = [float(lb), float(ub)]
 
-    # maxCurrent = 12 # Ampers
-    max_current = 2
+    # Define the initial state.
+    initial_configuration = [0.2,0,0.4, 0,0,0,1] + 4*[0.8,-1.6]
+    initial_velocity = (8+6)*[0,]
 
-    # maximum torques
-    max_torque = motor_torque_constant * max_current
+    q0 = zero(robot_model.nq)
+    q0[:] = np.asmatrix(initial_configuration).T
+    v0 = zero(robot_model.nv)
+    a0 = zero(robot_model.nv)
 
-    # maximum control one can send, here the control is the current.
-    max_control = max_current
 
-    # mapping between the ctrl vector in the device and the urdf indexes
-    urdf_to_dgm = (0,1,2,3,4,5,6,7)
 
-    # ctrl_manager_current_to_control_gain I am not sure what it does so 1.0.
-    ctrl_manager_current_to_control_gain=1.0
+class Solo12Config(SoloAbstract):
+    # name that is used by every other entities. I do not exactly why...
+    robot_family = "solo"
+    robot_name = "solo12"
+
+    # here we use the same urdf as for the quadruped but without the freeflyer
+    urdf_path = (
+        join(rospkg.RosPack().get_path("robot_properties_" + robot_family),
+             "urdf",
+             robot_name + ".urdf")
+    )
+
+    meshes_path = [
+      dirname(rospkg.RosPack().get_path("robot_properties_" + robot_family))
+    ]
+
+    yaml_path = (
+        join(rospkg.RosPack().get_path("robot_properties_" + robot_family),
+             "config",
+             "dgm_parameters.yaml")
+    )
+
+    # The inertia of a single blmc_motor
+    motor_inertia = 0.0000045
+
+    # The motor gear ratio
+    motor_gear_ration = 9.
+
+    # pinocchio model
+    robot_model = se3.buildModelFromUrdf(urdf_path,
+                                         se3.JointModelFreeFlyer())
+    robot_model.rotorInertia[6:] = motor_inertia
+    robot_model.rotorGearRatio[6:] = motor_gear_ration
+
+    base_name = robot_model.frames[2].name
+
+    # the number of motors, here they are the same as there are only revolute
+    # joints
+    nb_joints = robot_model.nv - 6
+
+    # mapping between the ctrl vector in the device and the urdf indexes
+    urdf_to_dgm = (0,1,2,3,4,5,6,7,8,9,10,11)
```

Comment:
range(12)?

Reply:
Changed this to `tuple(range(12))` and the above for `SoloConfig` to `tuple(range(6))`.